### PR TITLE
[MM-39060] - A/B Test: Reminder to Join Workspace email

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
This PR adds a new feature flag `ResendInviteEmailInterval` which determines the interval we send another invite to a user that has not yet joined a team. This interval is either 48 hours or 72 hours after the initial invite email.

#### Ticket Link
[MM-39060](https://mattermost.atlassian.net/browse/MM-39060)

#### Release Note

```release-note
NONE

```

### CODE REVIEW

File: jobs/resend_invitation_email/worker.go [hunk 1]
```diff
@@ -64,6 +67,76 @@ func (rseworker *ResendInvitationEmailWorker) JobChannel() chan<- model.Job {
...
+func (rseworker *ResendInvitationEmailWorker) DoJob_24_48(job *model.Job) {
+	elapsedTimeSinceSchedule, DurationInMillis_24, DurationInMillis_48, _ := rseworker.GetDurations(job)
+	systemValue, sysValErr := rseworker.App.Srv().Store.System().GetByName(model.NumberOfInviteEmailsSent)
+	if sysValErr != nil {
+		if _, ok := sysValErr.(*store.ErrNotFound); !ok {
+			mlog.Error("An error occurred while getting NUMBER_OF_INVITE_EMAILS_SENT from system store", mlog.String("worker", rseworker.name), mlog.Err(sysValErr))
+			return
+		}
+	}
+	if (elapsedTimeSinceSchedule > DurationInMillis_24) && (elapsedTimeSinceSchedule < DurationInMillis_48) && (systemValue.Value == "0") {
```
Comment (marianunez) [inline jobs/resend_invitation_email/worker.go:100]: Are we checking somewhere if the user already joined? Why do we check if it was previously sent?

Comment (AGMETEOR) [reply jobs/resend_invitation_email/worker.go:100]: Yes. The function `ResendEmails` calls `removeAlreadyJoined` to take care of that. The comparison `systemValue.Value == "0"` is for the job to not continously run the first branch of the if statement. I do this because the times are not exact. Without it, between `elapsedTimeSinceScshedule` and `DurationInMillis_48`, the first branch of the if statement can run several times. So, we want to mark it done.

File: jobs/resend_invitation_email/worker.go [hunk 2]
```diff
@@ -64,6 +67,76 @@ func (rseworker *ResendInvitationEmailWorker) JobChannel() chan<- model.Job {
...
+		}
+	}
+	if (elapsedTimeSinceSchedule > DurationInMillis_24) && (elapsedTimeSinceSchedule < DurationInMillis_72) && (systemValue.Value == "0") {
+		rseworker.ResendEmails(job)
+		rseworker.setNumResendEmailSent("1")
+	} else if elapsedTimeSinceSchedule > DurationInMillis_72 {
+		rseworker.ResendEmails(job)
+		rseworker.TearDown(job)
+	}
+}
```
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply jobs/resend_invitation_email/worker.go:125]: Yes. Pushed update...

**Comment by AGMETEOR** [comment]
@stevemudie the splitio flag for this is set. Currently to 48 hours.

File: api4/team.go [hunk 3]
```diff
@@ -1328,6 +1328,12 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
+		sysVar := &model.System{Name: model.NumberOfInviteEmailsSent, Value: "0"}
+		if sysValErr := c.App.Srv().Store.System().SaveOrUpdate(sysVar); sysValErr != nil {
+			c.Err = model.NewAppError("Api4.inviteUsersToTeam", "", nil, "Unable to save system value NUMBER_OF_INVITE_EMAIL_SENT", http.StatusInternalServerError)
+			return
```
Comment (marianunez) [inline api4/team.go:1334]: Should we still continue here if this fails? This setting is not critical to block sending out the initial invite

File: i18n/en.json [hunk 4]
```diff
@@ -1,4 +1,8 @@
 [
+  {
+    "id": "",
+    "translation": ""
+  },
```
Comment (marianunez) [inline i18n/en.json:5]: Is this change intended?

**Comment by marianunez** [comment]
@AGMETEOR one thing that I missed to bring up is that the reminder email design includes the "Reminder" text in the title (as below), if it's a simple change we can add to this PR or we can split this out to another PR 

Screen Shot 2021-11-04 at 1 12 42 PM

**Comment by AGMETEOR** [comment]
> @AGMETEOR one thing that I missed to bring up is that the reminder email design includes the "Reminder" text in the title (as below), if it's a simple change we can add to this PR or we can split this out to another PR
> 
> Screen Shot 2021-11-04 at 1 12 42 PM

Screenshot 2021-11-08 at 09 21 03
-----
Screenshot 2021-11-08 at 09 21 20

File: api4/team.go (hunk 3 above)
Comment (AGMETEOR) [reply api4/team.go]: Changed to just a log warning

File: i18n/en.json (hunk 4 above)
Comment (AGMETEOR) [reply i18n/en.json]: No it was not

**Review by nickmisasi — APPROVED** [review]

File: jobs/resend_invitation_email/worker.go (hunk 1 above)
Comment (marianunez) [reply jobs/resend_invitation_email/worker.go]: Makes sense, thanks for clarifying!

**Review by marianunez — COMMENTED** [review]
@AGMETEOR let me know when the telemetry is included here so I can do a final review. Commenting here to get it out of my review queue

**Review by marianunez — APPROVED** [review]
LGTM, thanks @AGMETEOR!

**Review by stevemudie — APPROVED** [review]
Worked as expected!

I received reminder emails on the designated time frames for both tests.

All set @AGMETEOR
